core/plugins/org.polarsys.capella.core.data.migration/src/org/polarsys/capella/core/data/migration/aird/AssociationCDBMigrationContributor.java

```
@@ -0,0 +1,376 @@
+
+    // changes
+    beginLabel.ifPresent(label -> migrateLabelPosition(label, bendpointsList, revBendpointsList));
+    middleLabel.ifPresent(label -> migrateLabelPosition(label, bendpointsList, revBendpointsList));
+    endLabel.ifPresent(label -> migrateLabelPosition(label, bendpointsList, revBendpointsList));
+
+    beginLabel.ifPresent(label -> label.setType(SiriusVisualIDRegistry.getType(DEdgeEndNameEditPart.VISUAL_ID)));
+    endLabel.ifPresent(label -> label.setType(SiriusVisualIDRegistry.getType(DEdgeBeginNameEditPart.VISUAL_ID)));
+
+    gmfEdge.setSource(targetView);
+    gmfEdge.setTarget(sourceView);
+    sourceViewTargetEdges.add(gmfEdge);
+    targetViewSourceEdges.add(gmfEdge);
+    sourceViewSourceEdges.remove(gmfEdge);
+    targetViewTargetEdges.remove(gmfEdge);
+
+    gmfEdge.setSourceAnchor(targetAnchor);
+    gmfEdge.setTargetAnchor(sourceAnchor);
+    bendpoints.setPoints(revBendpointsList);
+  }
+
+  private void migrateAssociation(DEdge dEdge) {
+    migrateDEdge(dEdge);
+    migrateGMFEdge(SiriusGMFHelper.getGmfEdge(dEdge));
+  }
+
+  private boolean needMigration(DEdge dEdge) {
+    if (dEdge.getTarget() instanceof Association association) {
+      Property sourceProp = InformationServices.getService().getAssociationSource(association);
+      Property targetProp = InformationServices.getService().getAssociationTarget(association);
+      AbstractType expectedSource = sourceProp.getAbstractType();
+      AbstractType expectedTarget = targetProp.getAbstractType();
+      if (dEdge.getSourceNode() instanceof DSemanticDecorator sourceNode
+          && dEdge.getTargetNode() instanceof DSemanticDecorator targetNode) {
+        EObject sourceObject = sourceNode.getTarget();
+        EObject targetObject = targetNode.getTarget();
+        return sourceObject == expectedTarget && targetObject == expectedSource;
+      } else {
+        // error
```

Maybe add something to some logs or throw an exception if the data are corrupted ?
Silently ignoring errors during migration is the best way to get strange bugs and data corruption visible months later

---

I added a log message.